app/assets/javascripts/discourse/app/components/d-editor.js

```diff
@@ -464,11 +463,17 @@ export default Component.extend(TextareaTextManipulation, {
         this.topicId,
         this.categoryId,
         this.siteSettings.max_oneboxes_per_post,
-        false,
-        true
+        /* refresh */ false,
+        /* offline */ true
       );
 
-      resolveCachedShortUrls(this.siteSettings, cookedElement);
+      resolveAllShortUrls(ajax, this.siteSettings, cookedElement);
```

Comment:
Needed to resolve **all** the short urls instead of just the **cached** ones to make sure we always convert short urls to a proper url before doing the morphing.